backend/fms_core/resources/project_link_sample.py

```diff
@@ -0,0 +1,104 @@
+import reversion
+from import_export.fields import Field
+from import_export.widgets import ForeignKeyWidget
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from ._generic import GenericResource
+from ._utils import skip_rows, add_columns_to_preview
+from ..models import Project, Sample, Container, SampleByProject
+from ..utils import get_normalized_str
+
+
+__all__ = ["ProjectLinkSampleResource"]
+
+ADD_ACTION = "ADD"
+REMOVE_ACTION = "REMOVE"
+
+class ProjectLinkSampleResource(GenericResource):
+    action = Field(column_name='Action')
+    project_name = Field(column_name='Project Name') 
+    sample_name = Field(column_name='Sample Name')
+    barcode = Field(column_name='Sample Container Barcode',
+                     widget=ForeignKeyWidget(Container, 'barcode'))
+    coordinates = Field(column_name='Sample Container Coord')
+
+    class Meta:
+        model = SampleByProject
+        export_order = ('action', 'project_name', 'sample_name', 'barcode', 'coordinates',)
+
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        skip_rows(dataset, 7)
+
+    def import_obj(self, obj, data, dry_run, **kwargs):
+        errors = {}
+        # Get field info
+        try:
+            super().import_obj(obj, data, dry_run, **kwargs)
+        except ValidationError as e:
+            errors.update(e.update_error_dict(errors).copy())
+        # Get sample object
+        try:
+            obj.sample = Sample.objects.get(name=data["Sample Name"],
+                                            container__barcode=data["Sample Container Barcode"],
+                                            coordinates=data["Sample Container Coord"])
+        except Sample.DoesNotExist:
+            obj.sample = None
+            errors["sample"] = ValidationError([f"No sample with name [{data['Sample Name']}] at those container coordinates."], code="invalid")
+        # Get project object
+        try:
+            obj.project = Project.objects.get(name=data["Project Name"])
+        except Project.DoesNotExist:
+            obj.project = None
+            errors["project"] = ValidationError([f"No project with name [{data['Project Name']}]"], code="invalid")
+
+        # Set object for deletion if REMOVE_ACTION
+        try:
+            link_exists = SampleByProject.objects.filter(sample=obj.sample, project=obj.project).exists()
+        except ObjectDoesNotExist:
+            link_exists = False
```

Comment:
Why not having a simpler logic such as if  queryset.exists(): blabla else: blablabla and re-arranging a bit the logic? I don't think there is a need to define an intermediate variable for this purpose

Reply:
@UlysseFG What do you think?

Reply:
Also, from the documentation I dont think exists() ever throws a ObjectDoesNotExist exception

Reply:
eheh i think i actually wrote this ... I am not sure this would be simpler ... it could be different but i don't see anything simpler (this is pretty simple). The intermediate variable could be removed but you would need to test the action at 2 different place in your code ... I do not know... up to you Sebastian really.

Reply:
But at least, if this is never going to throw an except, please remove the exception catching :)

Reply:
oh yeah you can most likely remove the exception handling for the exists()

Reply:
I suggest something along those lines:
![image](https://user-images.githubusercontent.com/1617741/131580313-ae4771bc-349b-4a1b-9ca2-2020e94ad3a8.png)


IMPORTANT: Also, to verify: is there any 'deleted' object handling when making queries? For example, if you query SampleByProject.objects.all() do the objects marked as 'deleted' still appear in the query? if not it should be managed in some of the model Manager or at a general place